benchmarks/nonlinear_channel_flow/simple_nonlinear.cc

```diff
@@ -0,0 +1,423 @@
+#include <aspect/simulator.h>
+#include <deal.II/grid/tria.h>
+#include <aspect/material_model/interface.h>
+#include <aspect/simulator_access.h>
+#include <aspect/newton.h>
+
+#include <iostream>
+
+#include <aspect/utilities.h>
+
+namespace aspect
+{
+  namespace MaterialModel
+  {
+    using namespace dealii;
+
+    /**
+     * A material model based on a simple power law rheology and
+     * implementing the derivatives needed for the Newton method.
+     *
+     * Power law equation to compute the viscosity $\eta$ per composition:
+     * $\eta = A * \dot\varepsilon_{II}^{\frac{1}{n}-1}$ where $A$ is the
+     * prefactor, $\dot\varepsion$ is the strain-rate, II indicates
+     * the square root of the second invariant defined as
+     * $\frac{1}{2} \dot\varepsilon_{ij} \dot\varepsilon_{ij}$, and
+     * $n$ is the stress exponent.
+     *
+     * The viscosities per composition are averaged using the
+     * utilities weighted p-norm function. The volume fractions are used
+     * as weights for the averaging.
+     *
+     * @ingroup MaterialModels
+     */
+    template <int dim>
+    class SimpleNonlinear : public MaterialModel::Interface<dim>, public ::aspect::SimulatorAccess<dim>
```

Comment:
I have no strong feelings where it should go. It used to be a separate material model, but Rene suggested that because it was unlikely to be used outside of benchmarking and tests it was better to put it in the benchmark folder, so I moved it there.